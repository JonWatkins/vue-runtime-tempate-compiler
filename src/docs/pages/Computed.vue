<template>
  <div class="page">
    <h1 id="top" class="display-4 mb-3">
      <router-link to="/guide/computed#top" class="header-anchor">
        <span class="header-anchor--hash">#</span>
        Computed
      </router-link>
    </h1>

    <p>
      Just like the <code>methods</code> the <code>computed</code> properties
      are available in the <code>vue-runtime-template-compiler</code>. The same
      as the other parent property you use them as if you are writing bindings
      in the parent's compiled template.
    </p>

    <pre class="language-html mb-4"><code>{{ example | trim }}</code></pre>

    <PageNav>
      <template v-slot:left>
        <router-link to="/guide/methods">
          <i class="fas fa-arrow-left"></i>
          Methods
        </router-link>
      </template>
      <template v-slot:right>
        <router-link to="/guide/props">
          Props
          <i class="fas fa-arrow-right"></i>
        </router-link>
      </template>
    </PageNav>
  </div>
</template>

<script>
import PageNav from '../components/PageNav.vue'

const example = `
<template>
  <runtime-template-compiler :template="template" \/>
<\/template>

<script>
import { RuntimeTemplateCompiler } from 'vue-runtime-template-compiler'

export default {
  components: {
    RuntimeTemplateCompiler
  },
  data: () => ({
    userName: 'Bob',
    template: \`<div>{{ greeting }}!<\/div>\`,
  }),
  computed: {
    greeting() {
      return \`Hello my name is \${this.userName}\`
    }
  }
}
<\/script>
`

export default {
  name: 'Computed',

  components: {
    PageNav
  },

  data() {
    return {
      example
    }
  }
}
</script>
