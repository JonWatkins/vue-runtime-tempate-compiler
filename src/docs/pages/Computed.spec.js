import { shallowMount } from '@vue/test-utils'
import Computed from './Computed.vue'
const { localVueInst } = global

let wrapper

describe('Computed', () => {
  beforeEach(() => {
    wrapper = shallowMount(Computed, {
      localVue: localVueInst
    })
  })

  it('is called', () => {
    expect(wrapper.name()).toBe('Computed')
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
