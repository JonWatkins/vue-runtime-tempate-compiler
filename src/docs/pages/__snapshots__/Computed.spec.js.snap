// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Computed render correctly 1`] = `
<div class="page">
  <h1 id="top" class="display-4 mb-3">
    <router-link-stub to="/guide/computed#top" tag="a" event="click" class="header-anchor"><span class="header-anchor--hash">#</span>
      Computed
    </router-link-stub>
  </h1>
  <p>
    Just like the <code>methods</code> the <code>computed</code> properties
    are available in the <code>vue-runtime-template-compiler</code>. The same
    as the other parent property you use them as if you are writing bindings
    in the parent's compiled template.
  </p> <pre class="language-html mb-4"><code>&lt;template&gt;
  &lt;runtime-template-compiler :template="template" /&gt;
&lt;/template&gt;

&lt;script&gt;
import { RuntimeTemplateCompiler } from 'vue-runtime-template-compiler'

export default {
  components: {
    RuntimeTemplateCompiler
  },
  data: () =&gt; ({
    userName: 'Bob',
    template: \`&lt;div&gt;{{ greeting }}!&lt;/div&gt;\`,
  }),
  computed: {
    greeting() {
      return \`Hello my name is \${this.userName}\`
    }
  }
}
&lt;/script&gt;</code></pre>
  <pagenav-stub></pagenav-stub>
</div>
`;
